control-plane/agents/core/src/controller/wrapper.rs

```diff
@@ -1105,17 +1108,36 @@ impl ClientOps for GrpcClientLocked {
             APIVersion::V0 => {
                 let rpc_nexus = self
                     .client_v0()?
-                    .create_nexus_v2(request.to_rpc())
+                    .create_nexus_v2(v0_conversion::to_rpc(request))
                     .await
                     .context(GrpcRequestError {
                         resource: ResourceKind::Nexus,
                         request: "create_nexus",
                     })?;
-                let nexus = rpc_nexus_to_agent(&rpc_nexus.into_inner(), &request.node)?;
+                let nexus = v0_rpc_nexus_to_agent(&rpc_nexus.into_inner(), &request.node)?;
                 Ok(nexus)
             }
             APIVersion::V1 => {
-                unimplemented!()
+                let rpc_nexus = self
+                    .client_v1()?
+                    .nexus()
+                    .create_nexus(v1_conversion::to_rpc(request))
+                    .await
+                    .context(GrpcRequestError {
+                        resource: ResourceKind::Nexus,
+                        request: "create_nexus",
+                    })?;
```

Comment:
We need to handle the idempotency differently now - If the nexus exists the create fails on the v1 api.

Reply:
Ohh yes! Will handle those.